types/frontmatter.ts

```diff
@@ -0,0 +1,63 @@
+export type Frontmatter = {
+    /**
+     * Title.
+     * Required.
+     */
+    title: string;
+
+    /**
+     * Intro.
+     * Optional.
+     */
+    intro?: string;
+
+    /**
+     * Indicates whether the `title` in the frontmatter can differ
+     * from the `title` in the filename.
+     * Optional. Default is `false`.
+     */
+    allowTitleToDifferFromFilename?: boolean;
+
+    /**
+     * Date (YYYY-MM-DD).
+     * Required.
+     */
+    date: string;
+
+    /**
+     * Array of authors (array of `id` of `Author`).
+     * Required (with at least one author).
+     */
+    authors: string[];
+
+    /**
+     * YouTube video URL.
+     * Optional.
+     */
+    featuredVideo?: string;
```

Comment:
```suggestion
    featuredVideo?: URL;
```
if it's always a URL then why not to enforce it?

Reply:
I agree. Done.